### PR TITLE
Get master key

### PR DESCRIPTION
**Pull request by aljones15**
just takes the getMasterKey method from veres-wallet utils.js file and moves the getMasterKey method over.
Service and plugin are both optional parameters.
Tested this and works just fine on Register and Login in veres-wallet.

### CODE REVIEW

File: AccountMasterKey.js [hunk 1]
```diff
@@ -178,6 +179,30 @@ export class AccountMasterKey {
...
+  /**
+   * Gets the master key from either the cache or a secret.
+   *
+   * @param {Object} options - Options for generating a master key.
+   * @param {string} options.accountId - An account id.
+   * @param {string} [options.secret = null]
+   * - A secret that is used to generate a key.
+   * @param {string} [options.plugin = 'ssm-v1']
+   * - The plugin type for the service.
+   * @param {Object} [options.Service = KmsService]
```
Comment (dlongley) [inline AccountMasterKey.js]: This should pass the instance, not the class.

File: AccountMasterKey.js [hunk 2]
```diff
@@ -178,6 +179,30 @@ export class AccountMasterKey {
     const signer = await _signerFromSeed({seed});
     return new AccountMasterKey({accountId, signer, kmsService, kmsPlugin});
   }
+  /**
```
Comment (dlongley) [inline AccountMasterKey.js]: Needs empty line whitespace above this to break things up.

File: AccountMasterKey.js [hunk 3]
```diff
@@ -178,6 +179,30 @@ export class AccountMasterKey {
...
+   * Gets the master key from either the cache or a secret.
+   *
+   * @param {Object} options - Options for generating a master key.
+   * @param {string} options.accountId - An account id.
+   * @param {string} [options.secret = null]
+   * - A secret that is used to generate a key.
+   * @param {string} [options.plugin = 'ssm-v1']
```
Comment (dlongley) [inline AccountMasterKey.js]: This shouldn't have any default here.

File: AccountMasterKey.js [hunk 4]
```diff
@@ -178,6 +179,30 @@ export class AccountMasterKey {
...
+   * - The plugin type for the service.
+   * @param {Object} [options.Service = KmsService]
+   * - The service to use to get the key.
+   *
+   * @returns {AccountMasterKey} A key.
+   */
+  static getMasterKey(
```
Comment (dlongley) [inline AccountMasterKey.js]: This naming convention doesn't match the rest of the class. Perhaps `fromCache` should just be updated to optionally accept a `secret` and to make passing the `kmsService` optional?

**Review by dlongley — CHANGES_REQUESTED** [review]

File: AccountMasterKey.js (hunk 4 above)
Comment (aljones15) [reply AccountMasterKey.js]: or perhaps utils.js should just call explicitly on fromCache or fromSecret and not use this method at all?

File: AccountMasterKey.js [hunk 5]
```diff
@@ -158,24 +159,32 @@ export class AccountMasterKey {
...
    *   operations.
    * @param {string} options.kmsPlugin - The ID of the KMS plugin to use.
+   * @param {string} [options.secret = null]
+   * - A secret that is used to generate a key.
    *
    * @returns {Promise<AccountMasterKey>} The new AccountMasterKey instance
    *   or `null` if no cached key for `accountId` could be loaded.
    */
-  static async fromCache({accountId, kmsService, kmsPlugin}) {
+  static async fromCache({accountId, service, kmsPlugin, secret = null}) {
```
Comment (dlongley) [inline AccountMasterKey.js]:
```suggestion
  static async fromCache({accountId, kmsService = new KmsService(), kmsPlugin, secret = null}) {
```

File: AccountMasterKey.js [hunk 6]
```diff
@@ -158,24 +159,32 @@ export class AccountMasterKey {
    * @param {Object} options - The options to use.
    * @param {string} options.accountId - The ID of the account associated with
    *   this master key.
-   * @param {Object} options.kmsService - The kmsService to use to perform key
+   * @param {Object} [options.service] - The kmsService to use to perform key
```
Comment (dlongley) [inline AccountMasterKey.js]: Need to keep this the same or it's a breaking change.

```suggestion
   * @param {Object} [options.kmsService] - The kmsService to use to perform key
```

File: AccountMasterKey.js [hunk 7]
```diff
@@ -158,24 +159,32 @@ export class AccountMasterKey {
...
    *   operations.
    * @param {string} options.kmsPlugin - The ID of the KMS plugin to use.
+   * @param {string} [options.secret = null]
+   * - A secret that is used to generate a key.
```
Comment (dlongley) [inline AccountMasterKey.js]: Can we move this up a line or line-break it differently?

File: AccountMasterKey.js [hunk 8]
```diff
@@ -158,24 +159,32 @@ export class AccountMasterKey {
...
    *
    * @returns {Promise<AccountMasterKey>} The new AccountMasterKey instance
    *   or `null` if no cached key for `accountId` could be loaded.
    */
-  static async fromCache({accountId, kmsService, kmsPlugin}) {
+  static async fromCache({accountId, service, kmsPlugin, secret = null}) {
+    const kmsService = service || new KmsService();
```
Comment (dlongley) [inline AccountMasterKey.js]:
```suggestion
```

File: AccountMasterKey.js [hunk 9]
```diff
@@ -158,24 +159,32 @@ export class AccountMasterKey {
...
     if(typeof localStorage === 'undefined') {
       return null;
     }
 
     const seed = await _seedCache.get({accountId});
     if(!seed) {
+      if(secret !== null) {
```
Comment (dlongley) [inline AccountMasterKey.js]: I think we need to put this outside of the seed check or we could load the wrong account master key (a stale one that doesn't match the given `secret` from the cache).

File: AccountMasterKey.js [hunk 10]
```diff
@@ -158,24 +159,32 @@ export class AccountMasterKey {
...
+        return AccountMasterKey.fromSecret(
+          {secret, accountId, kmsService, kmsPlugin});
+      }
       return null;
     }
 
     const signer = await _signerFromSeed({seed});
+
```
Comment (dlongley) [inline AccountMasterKey.js]:
```suggestion
```

**Review by dlongley — CHANGES_REQUESTED** [review]

File: AccountMasterKey.js (hunk 9 above)
Comment (aljones15) [reply AccountMasterKey.js]: moved it outside of the seed check.

**Review by dlongley — APPROVED** [review]

**Pull request merged**